src/main/webapp/view/discussion/discussion.js

```diff
@@ -12,26 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import MarkAnsweredUtil from './mark-answered-util.js';
+
 export const COMMENT_TYPE_REPLY = 'REPLY';
 export const COMMENT_TYPE_QUESTION_UNANSWERED = 'QUESTION_UNANSWERED';
 export const COMMENT_TYPE_QUESTION_ANSWERED = 'QUESTION_ANSWERED';
 export const COMMENT_TYPE_NOTE = 'NOTE';
 
+/**
+ * Provides information on a comment type. For instance: its name,
+ * display styles, and if it can be changed.
+ * @typedef {Object} CommentTypeInfo
+ * @property {string} name The display name of the type.
+ * @property {!Array<string>} badgeStyles List of classes to add to the type
+ *     tag. TODO: Rename to tagStyles.
+ * @property {boolean} isQuestion If the type is a question, meaning the
+ *     answered status can be changed.
+ * @property {string=} oppositeType The opposite type of comment, for
+ *     question types.
+ * @property {string=} updateTypeText Text to display on a button to change a
+ *     comment's type to this type, for question types.
+ * @property {function(Object):void=} updateTypeFunction A function to set a
+ *     comment to this type, for question types.
+ */
+
+/** @type {Object.<string, CommentTypeInfo>} */
```

Comment:
Why is this a second JsDoc?

Reply:
This one annotates the type of the `COMMENT_TYPES` object/map, the previous one defines the type `CommentTypeInfo`.

Reply:
Ah I see. It is a JsDoc for the type.